alvr/common/src/average.rs

```diff
@@ -38,3 +39,40 @@ impl SlidingWindowAverage<Duration> {
         self.history_buffer.iter().sum::<Duration>() / self.history_buffer.len() as u32
     }
 }
+
+impl SlidingWindowAverage<Vec2> {
+    pub fn get_average(&self) -> Vec2 {
+        self.history_buffer.iter().sum::<Vec2>() / self.history_buffer.len() as f32
+    }
+}
+
+pub struct AngleSlidingWindowAverage {
+    history_buffer: VecDeque<Vec2>,
+    max_history_size: usize,
+}
+
+impl AngleSlidingWindowAverage {
+    pub fn new(initial_value: f32, max_history_size: usize) -> Self {
+        Self {
+            history_buffer: [Vec2::from_angle(initial_value)].into_iter().collect(),
+            max_history_size,
+        }
```

Comment:
This goes for the main impl too: It should reserve here, that way it only has to allocate once.

Reply:
The best I can do is initialize with capacity